### PR TITLE
test(common,logger,metrics): run e2e tests for metrics in both Node runtimes

### PR DESCRIPTION
**Pull request by ijemmy**
## Description of your changes

Changes in this PR.
- refactor metrics e2e tests to make stack and resource name unique (so that they can be run in parallel)
- refactor metrics e2e tests to be consistent with logger and move common code to `commons`
- run e2e tests on both Node12 and Node14 runtimes

### How to verify this change

Run `npm run lerna-test:e2e` at the root.  

Every test should pass and metrics tests should be on both node14 and node12

### Related issues, RFCs
#497 

### PR status

***Is this ready for review?:*** NO  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/commons/src/index.ts
Comment (dreamorosi) [inline packages/commons/src/index.ts:3]: If we export it here, does it mean the module gets included in the published package? If that's the case maybe we don't want that

File: packages/commons/jest.config.js
Comment (dreamorosi) [inline packages/commons/jest.config.js:23]:
```suggestion
    * e2eUtils.ts contains helper methods that simplify several calls to CDK and SDK interface.
```

File: packages/commons/src/tests/e2e/e2eUtils.ts
Comment (dreamorosi) [inline packages/commons/src/tests/e2e/e2eUtils.ts:1]: The reason why we ignore this file from coverage is the one you mention in the comment at `packages/commons/jest.config.js`? 
If I understand it correctly we are saying that it's not worth it to write unit tests for the files because by running the integration tests we are already "testing" it.

The only minor issue I see is that at the moment we are not enforcing e2e tests on merge, but I see your point and agree on the low value of writing the unit tests for this.

File: packages/commons/src/index.ts
Comment (ijemmy) [reply packages/commons/src/index.ts:3]: Good point. I just saw Florian's PR that he import directly (e.g. `../../../commons/tests/utils/cdk-cli`) . This makes more sense as the code is internal.

File: packages/commons/src/tests/e2e/e2eUtils.ts
Comment (ijemmy) [reply packages/commons/src/tests/e2e/e2eUtils.ts:1]: If forgot to remove this, the ignore in `jest.config.js` is enough.

Comment (ijemmy) [reply packages/commons/src/tests/e2e/e2eUtils.ts:1]: But yes, that's why I didn't write unit tests for the file.  If I don't ignore it, it'll drop our coverage percentage from 100%.

**Review by dreamorosi — COMMENTED** [review]
Thanks a lot for the PR, I like the direction you're giving to the integration tests. It'll make them much easier to maintain moving forward.

My only concern relates to exporting the e2e utilities in the `commons package`. This package is depended and bundled by all the other utilities and as such it will be included in the deployment package of each function that uses Powertools. 

While I think there's definitely value in providing test utilities (see #445), I don't know if it should be in `commons` and in general, if it's code that should be deployed in Lambda.

What's your take?

**Comment by ijemmy** [comment]
> Thanks a lot for the PR, I like the direction you're giving to the integration tests. It'll make them much easier to maintain moving forward.
> 
> My only concern relates to exporting the e2e utilities in the `commons package`. This package is depended and bundled by all the other utilities and as such it will be included in the deployment package of each function that uses Powertools.
> 
> While I think there's definitely value in providing test utilities (see #445), I don't know if it should be in `commons` and in general, if it's code that should be deployed in Lambda.
> 
> What's your take?

I totally agree with you. We'll unnecessarily increase package size. I'll change this. 
Found that I modify quite a few files with #676 . I'll wait until that one is merge first, resolve conflict, and change this.

File: packages/metrics/package.json
Comment (flochaz) [inline packages/metrics/package.json:16]: wrong copy/paste. And I don't think you need to specify the module after e2e since it's a module specific package.json so it won't see otheres e2e tests anyway

File: packages/metrics/package.json
Comment (flochaz) [inline packages/metrics/package.json:17]: same

**Review by flochaz — CHANGES_REQUESTED** [review]

File: packages/metrics/package.json
Comment (ijemmy) [reply packages/metrics/package.json:16]: Good catch!

File: packages/metrics/package.json
Comment (ijemmy) [reply packages/metrics/package.json:17]: Good catch!

File: packages/metrics/package.json
Comment (ijemmy) [reply packages/metrics/package.json:16]: Good catch!

File: packages/metrics/package.json
Comment (ijemmy) [reply packages/metrics/package.json:17]: Good catch!

**Review by flochaz — APPROVED** [review]

**Review by dreamorosi — APPROVED** [review]

**Pull request merged**
